qa/workunits/rados/test_librados_build.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash -ex
+#
+# 1. download example librados programs, and their Makefile, from git.ceph.com to the current working directory
+# 2. run "make all-system"
+# 3. check that expected binaries are created
+#
+# The script assumes all dependencies - e.g. wget, make, gcc, librados-devel, libradosstriper-devel, etc.
+# - are already installed.
+#
+
+trap cleanup EXIT
+
+SOURCES="hello_radosstriper.cc
+hello_world_c.c
+hello_world.cc
+Makefile
+"
+BINARIES="hello_radosstriper_cpp
+hello_world_c
+hello_world_cpp
+"
+BINARIES_TO_RUN="hello_world_c
+hello_world_cpp
+"
+DL_PREFIX="http://git.ceph.com/?p=ceph.git;a=blob_plain;f=examples/librados/"
+DESTDIR=$(pwd)
+
+function cleanup () {
+    for f in "$BINARIES$SOURCES" ; do
+        rm -f $f
+    done
+}
+
+function download () {
+    for s in $SOURCES ; do
+        curl --progress-bar --output $s ${DL_PREFIX}$s
+    done
+}
+
+function verify_download () {
+    for s in $SOURCES ; do
+        test -f $s
+    done
+}
+
+function check_binaries () {
+    for b in $BINARIES ; do
+        file $b
+        test -f $b
+    done
+}
+
+function run_binaries () {
+    for b in $BINARIES_TO_RUN ; do
+        ./$b
+    done
+}
+
+pushd $DESTDIR
+download
+verify_download
+make all-system 
+check_binaries
+run_binaries
+cleanup
```

Comment:
drop this line (`cleanup` is called via `trap` on EXIT)

Reply:
Did you want to do that before merging?